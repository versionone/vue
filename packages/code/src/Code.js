import React from 'react';
import SyntaxHighlighter from 'react-syntax-highlighter/prism';
import virtualizedRenderer from 'react-syntax-highlighter-virtualized-renderer';
import * as PropTypes from 'prop-types';
import supportedLanguages from './supportedLanguages';

export default function Code({
  children,
  language,
  lineNumberStyle,
  shouldShowLineNumbers,
  startingLineNumber,
}) {
  return (
    <SyntaxHighlighter
      language={language}
      lineNumberStyle={lineNumberStyle}
      renderer={virtualizedRenderer()}
      showLineNumbers={shouldShowLineNumbers}
      startingLineNumber={startingLineNumber}
      wrapLines
    >
      {children}
    </SyntaxHighlighter>
  );
}
Code.propTypes = {
  children: PropTypes.string,
  lineNumberStyle: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
  language: PropTypes.oneOf(supportedLanguages),
  shouldShowLineNumbers: PropTypes.bool,
  startingLineNumber: PropTypes.number,
};
Code.defaultProps = {
  lineNumberStyle: {},
  startingLineNumber: 1,
};
