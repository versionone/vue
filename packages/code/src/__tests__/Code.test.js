import React from 'react';
import { mount, shallow } from 'enzyme';
import Code from '../Code';

test('a code string without a language is rendered with a generic language formatting', () => {
  expect(shallow(<Code>ls -a</Code>)).toMatchSnapshot();
});

test('component can be configured to render a specific language', () => {
  expect(shallow(<Code language="bash">ls -a</Code>)).toMatchSnapshot();
});

test('optionally render line numbers', () => {
  expect(
    shallow(
      <Code language="bash" shouldShowLineNumbers>
        ls -a
      </Code>,
    ),
  ).toMatchSnapshot();
});
test('optionally start at line numbers at a given number', () => {
  expect(
    shallow(
      <Code language="bash" shouldShowLineNumbers startingLineNumber={10}>
        ls -a
      </Code>,
    ),
  ).toMatchSnapshot();
});

test('defaults to starting line number at 1', () => {
  expect(
    shallow(
      <Code language="bash" shouldShowLineNumbers>
        ls -a
      </Code>,
    ),
  ).toMatchSnapshot();
});

test('can apply a style object to each line', () => {
  expect(
    shallow(<Code lineNumberStyle={{ color: 'blue' }}>ls -a</Code>),
  ).toMatchSnapshot();
});
