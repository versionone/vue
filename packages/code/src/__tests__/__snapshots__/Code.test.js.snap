// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a code string without a language is rendered with a generic language formatting 1`] = `
<SyntaxHighlighter
  lineNumberStyle={Object {}}
  renderer={[Function]}
  startingLineNumber={1}
  wrapLines={true}
>
  ls -a
</SyntaxHighlighter>
`;

exports[`can apply a style object to each line 1`] = `
<SyntaxHighlighter
  lineNumberStyle={
    Object {
      "color": "blue",
    }
  }
  renderer={[Function]}
  startingLineNumber={1}
  wrapLines={true}
>
  ls -a
</SyntaxHighlighter>
`;

exports[`can apply styles via a callback for each line 1`] = `
<SyntaxHighlighter
  lineNumberStyle={[Function]}
  startingLineNumber={1}
  wrapLines={true}
>
  ls -a
</SyntaxHighlighter>
`;

exports[`component can be configured to render a specific language 1`] = `
<SyntaxHighlighter
  language="bash"
  lineNumberStyle={Object {}}
  renderer={[Function]}
  startingLineNumber={1}
  wrapLines={true}
>
  ls -a
</SyntaxHighlighter>
`;

exports[`defaults to starting line number at 1 1`] = `
<SyntaxHighlighter
  language="bash"
  lineNumberStyle={Object {}}
  renderer={[Function]}
  showLineNumbers={true}
  startingLineNumber={1}
  wrapLines={true}
>
  ls -a
</SyntaxHighlighter>
`;

exports[`optionally render line numbers 1`] = `
<SyntaxHighlighter
  language="bash"
  lineNumberStyle={Object {}}
  renderer={[Function]}
  showLineNumbers={true}
  startingLineNumber={1}
  wrapLines={true}
>
  ls -a
</SyntaxHighlighter>
`;

exports[`optionally start at line numbers at a given number 1`] = `
<SyntaxHighlighter
  language="bash"
  lineNumberStyle={Object {}}
  renderer={[Function]}
  showLineNumbers={true}
  startingLineNumber={10}
  wrapLines={true}
>
  ls -a
</SyntaxHighlighter>
`;
